slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Manager.java

```diff
@@ -0,0 +1,142 @@
+import tech.pegasys.web3signer.slashingprotection.interchange.model.Metadata;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.tuweni.bytes.Bytes;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+
+public class InterchangeV5Manager implements InterchangeManager {
+
+  private static final int FORMAT_VERSION = 5;
+
+  private final Jdbi jdbi;
+  private final ValidatorsDao validatorsDao;
+  private final SignedBlocksDao signedBlocksDao;
+  private final SignedAttestationsDao signedAttestationsDao;
+  private final ObjectMapper mapper;
+
+  public InterchangeV5Manager(
+      final Jdbi jdbi,
+      final ValidatorsDao validatorsDao,
+      final SignedBlocksDao signedBlocksDao,
+      final SignedAttestationsDao signedAttestationsDao,
+      final ObjectMapper mapper) {
+    this.jdbi = jdbi;
+    this.validatorsDao = validatorsDao;
+    this.signedBlocksDao = signedBlocksDao;
+    this.signedAttestationsDao = signedAttestationsDao;
+    this.mapper = mapper;
+  }
+
+  @Override
+  public void export(final OutputStream out) throws IOException {
+    try (final JsonGenerator jsonGenerator = mapper.getFactory().createGenerator(out)) {
+      jsonGenerator.writeStartObject();
+
+      final Metadata metadata = new Metadata(FORMAT_VERSION, Bytes.fromHexString("FFFFFFFF"));
+
+      jsonGenerator.writeFieldName("metadata");
+      mapper.writeValue(jsonGenerator, metadata);
+
+      jsonGenerator.writeArrayFieldStart("data");
+      populateInterchangeData(jsonGenerator);
+      jsonGenerator.writeEndArray();
+
+      jsonGenerator.writeEndObject();
+    }
+  }
+
+  private void populateInterchangeData(final JsonGenerator jsonGenerator) {
+    jdbi.useTransaction(
+        h ->
+            validatorsDao
+                .findAllValidators(h)
+                .forEach(
+                    validator -> {
+                      try {
+                        populateValidatorRecord(h, validator, jsonGenerator);
+                      } catch (IOException e) {
+                        throw new RuntimeException("Failed to construct a validator entry in json");
+                      }
+                    }));
+  }
+
+  private void populateValidatorRecord(
+      final Handle h, final Validator validator, final JsonGenerator jsonGenerator)
+      throws IOException {
+    jsonGenerator.writeStartObject();
+    jsonGenerator.writeStringField("pubkey", validator.getPublicKey().toHexString());
+    writeBlocks(h, validator, jsonGenerator);
+    writeAttestations(h, validator, jsonGenerator);
+    jsonGenerator.writeEndObject();
+  }
+
+  private void writeBlocks(
+      final Handle h, final Validator validator, final JsonGenerator jsonGenerator)
+      throws IOException {
+    jsonGenerator.writeArrayFieldStart("signed_blocks");
+
+    signedBlocksDao
```

Comment:
Can we add some logging or system out possibly so user can see progress that we currently exporting this validator. Could take a long time to do the export so some verbose output or the option to do so would be nice.